Packs/CiscoEmailSecurity/Integrations/CiscoEmailSecurity/CiscoEmailSecurity.py

```diff
@@ -0,0 +1,629 @@
+import demistomock as demisto
+from CommonServerPython import *
+
+import urllib3
+import traceback
+from typing import Any, Dict
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+MAX_MESSAGES_TO_GET = 20
+SAFELIST_AND_BLOCKLIST_TO_FILTER_URL = {
+    "safelist": "/sma/api/v2.0/quarantine/safelist",
+    "blocklist": "/sma/api/v2.0/quarantine/blocklist"
+}
+
+
+class Client(BaseClient):
+
+    def __init__(self, params):
+        self.username = params.get('api_username')
+        self.password = params.get('api_password')
+        super().__init__(base_url=params.get('base_url'), verify=not params.get('insecure', False),
+                         ok_codes=tuple(), proxy=params.get('proxy', False))
+
+        # self._token_base64 = self._generate_base64_token()
+        # self._headers = {'Authorization': 'Basic ' + self._token_base64}
+
+        self._jwt_token = self._generate_jwt_token()
+        self._headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'jwtToken': self._jwt_token
+        }
+
+    def http_request(self, method, url_suffix, full_url=None, headers=None, json_data=None, params=None, data=None,
+                     files=None, timeout=10, ok_codes=None, return_empty_response=False, auth=None):
+
+        return super()._http_request(method=method, url_suffix=url_suffix, full_url=full_url, headers=headers,
+                                     json_data=json_data, params=params, data=data, files=files, timeout=timeout,
+                                     ok_codes=ok_codes, return_empty_response=return_empty_response, auth=auth)
+
+    def _generate_base64_token(self) -> str:
+        basic_authorization_to_encode = f'{self.username}:{self.password}'
+        basic_authorization = base64.b64encode(basic_authorization_to_encode.encode('ascii')).decode('utf-8')
+        return basic_authorization
+
+    def _generate_jwt_token(self) -> str:
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json'
+        }
+        data = {
+            "data":
+                {
+                    "userName": base64.b64encode(self.username.encode('ascii')).decode('utf-8'),
+                    "passphrase": base64.b64encode(self.password.encode('ascii')).decode('utf-8')
+                }
+        }
+        output = self.http_request('POST', '/sma/api/v2.0/login', json_data=data, headers=headers)
+        jwt_token = output.get('data').get('jwtToken')
+        return jwt_token
+
+    def list_report(self, url_suffix) -> Dict[str, Any]:
+        return self.http_request(
+            method='GET',
+            url_suffix=url_suffix,
+        )
+
+    def list_messages(self, url_suffix) -> Dict[str, Any]:
```

Comment:
the url suffix shouldn't be as part of the request, you should have it built-in.